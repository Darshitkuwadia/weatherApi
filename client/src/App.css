.form {
  padding-top: 40px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.label {
  margin-bottom: 20px;
  font-size: 20px;
  color: var(--color-black);
}

.input-text {
  width: 300px;
  padding: 10px;
  font-size: 20px;
  border-width: 2px;
  border-color: var(--color-grey2);
  background-color: var(--color-white);
  border-style: solid;
  border-radius: 0px;
}

.input:focus {
  outline:none;
}

.input-button[type=submit]{
  background-color: var(--color-brand);
  border: none;
  color: var(--color-white);
  padding: 10px 32px;
  font-size: 20px;
  text-decoration: none;
  margin: 20px 20px;
  cursor: pointer;
  border: 2px solid transparent;
}

.input-button[type=submit]:focus,
.input-button[type=submit]:hover{
  background-color: var(--color-white);
  color: var(--color-brand);
  border: 2px solid var(--color-brand);
}

.input-error {
  font-size: 15px;
  color: var(--color-error);
}