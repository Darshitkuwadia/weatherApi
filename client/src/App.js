import React, { useState } from 'react';

import Layout from './components/Layout'
import WeatherTable from './components/WeatherTable'

import './App.css';

function App() {
  // define states of APP
  const [inputCityValue, setInputCityValue] = useState('');
  const [citiesWeatherData, setCitiesWeatherData] = useState({});
  const [error, setError] = useState('');

  function handleOnSubmit(event) {
    event.preventDefault();

    const fetchData = async () => {
      let response = await fetch(`/weather/${inputCityValue}`);
      // Handle response error here
      if(response.status !== 200) {
        let error = await response.json();
        setError(error);
      } else { // fetch data from here
        let data = await response.json();
        setError('');
        data['city'] = inputCityValue;
        setCitiesWeatherData({ ...citiesWeatherData, [inputCityValue]: data });
      }
    };

    // fetch data if inputCityValue is not empty
    if(!(inputCityValue === '')) { 
      fetchData();
    } else {
      setError('Input data is empty');
    }
  }

  // update inputCityValue on input onBlur  
  const handleOnBlur = (event) => {
    setInputCityValue(event.target.value.trim().toLowerCase());
  }

  // render multiuple WeatherTableRows from citiesWeatherData
  let WeatherTableRowElements = Object.keys(citiesWeatherData).map((cityWeatherData, i) => {
    return (
      <WeatherTable.Row
        key={i}
        city={citiesWeatherData[cityWeatherData].city}
        weatherIcon={citiesWeatherData[cityWeatherData].currently.icon}
        time={citiesWeatherData[cityWeatherData].currently.time}
        summary={citiesWeatherData[cityWeatherData].currently.summary}
        temperature={citiesWeatherData[cityWeatherData].currently.temperature}
      />
    )
  });

  return (
    <div className="App">
      <Layout
        title="[ iWeather ]" 
        subTitle="An application to find weather report of city"
      >
        <form 
          id="weatherForm"
          className="form" 
          onSubmit={handleOnSubmit}
        >
          <label className="label">
            Type City Name Here:
          </label>
          <input 
            id="cityName" 
            name="cityName" 
            className="input-text" 
            type="text" 
            onBlur={handleOnBlur}
            placeholder="Browse for your location"
          />          
          {error &&
            <p className="input-error">{error}</p>
          }
          <input
            className="input-button" 
            type="submit" 
            value="Search" 
          />
        </form>

        {(Object.keys(citiesWeatherData).length !== 0) && 
          <WeatherTable>
            {WeatherTableRowElements}
          </WeatherTable>
        }
      </Layout>
    </div>
  )
}

export default App