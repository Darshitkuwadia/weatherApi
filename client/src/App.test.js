import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('should render the App Component correctly', () => { 
    let app = shallow(
      <App /> 
    )  
    expect(app).toMatchSnapshot();
  })
  describe('[onSearch]', () => {
    it('is invoked when clicking the search Button', () => {
      const appOnClick = shallow(
        <App />
      )
      const event = { preventDefault: () => {} }
      jest.spyOn(event, 'preventDefault')
      appOnClick.find('#weatherForm').simulate('submit', event);
    })
  })
}); 