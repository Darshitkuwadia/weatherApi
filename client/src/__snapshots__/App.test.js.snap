// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App Component correctly 1`] = `
<div
  className="App"
>
  <Layout
    subTitle="An application to find weather report of city"
    title="[ iWeather ]"
  >
    <form
      className="form"
      id="weatherForm"
      onSubmit={[Function]}
    >
      <label
        className="label"
      >
        Type City Name Here:
      </label>
      <input
        className="input-text"
        id="cityName"
        name="cityName"
        onBlur={[Function]}
        placeholder="Browse for your location"
        type="text"
      />
      <input
        className="input-button"
        type="submit"
        value="Search"
      />
    </form>
  </Layout>
</div>
`;
